test/integration/DCAHubCompanion/position-migrator.spec.ts

```diff
@@ -110,11 +114,36 @@ contract('PositionMigrator', () => {
     when(title, () => {
       let sourcePositionId: BigNumber, targetPositionId: BigNumber;
       let swappedBalance: BigNumber, unswappedBalance: BigNumber;
-      let tx: TransactionResponse;
       given(async () => {
         ({ positionId: sourcePositionId, swappedBalance, unswappedBalance } = await depositInHubAndSwap(sourceHub()));
         const signature = await generateSignature(sourceHub(), positionOwner, sourcePositionId);
-        tx = await migrator.migrate(sourceHub().address, sourcePositionId, signature, targetHub().address);
+        const position = await sourceHub().userPosition(sourcePositionId);
+        const { data: permissionData } = await DCAHubCompanion.populateTransaction.permissionPermit(
+          await sourceHub().permissionManager(),
+          signature.permissions,
+          sourcePositionId,
+          signature.deadline,
+          signature.v,
+          signature.r,
+          signature.s
+        );
+        const { data: terminateData } = await DCAHubCompanion.populateTransaction.terminate(
+          sourceHub().address,
+          sourcePositionId,
+          DCAHubCompanion.address,
+          positionOwner.address
+        );
+        const { data: depositData } = await DCAHubCompanion.populateTransaction.depositWithBalanceOnContract(
+          targetHub().address,
+          position.from,
+          position.to,
+          position.swapsLeft,
+          position.swapInterval,
+          positionOwner.address,
+          [],
+          []
+        );
+        const tx = await DCAHubCompanion.multicall([permissionData!, terminateData!, depositData!]);
```

Comment:
That moment when you only need to change this because you've done an amazing test.